journals/apps/api/v1/views.py

```diff
@@ -62,3 +64,31 @@ def _revoke_access(self, order_number):
                     access_record.journal,
                     access_record.order_number)
         return HttpResponse()
+
+
+class UserPageVisitViewSet(viewsets.ModelViewSet):
+    """API for UserPageVisit model"""
+    lookup_field = 'user_id'
+    queryset = UserPageVisit.objects.all()
+    serializer_class = UserPageVisitSerializer
+    permission_classes = (IsAdminUser,)
+
+    def retrieve(self, request, *args, **kwargs):   # pylint: disable=unused-argument,useless-suppression
+        """
+        Returns the user's visited pages.
+        page_id is optional argument.
+        """
+        user_id = kwargs['user_id']
+        page_id = self.request.GET.get('page_id')
+        queryset = self.queryset.filter(user_id=user_id)
+        if page_id:
+            queryset = queryset.filter(page_id=page_id)
+        return Response(self.serializer_class(queryset, many=True).data)
+
+    @detail_route(url_path='lastvisit')
+    def user_last_visited_page(self, request, user_id):  # pylint: disable=unused-argument,useless-suppression
+        """
+        Returns the user's last visited page.
+        """
+        queryset = self.queryset.filter(user_id=user_id).order_by("-visited_at")[0]
```

Comment:
This can be changed to something like this 
```
queryset = self.queryset.filter(user_id=user_id).first()
```
since we have ordering in-place on base queryset.